internal/statesync/reactor.go

```diff
@@ -779,7 +779,7 @@ func (r *Reactor) handleParamsMessage(envelope p2p.Envelope) error {
 		if sp, ok := r.stateProvider.(*stateProviderP2P); ok {
 			select {
 			case sp.paramsRecvCh <- cp:
-			default:
+			case <-time.After(time.Second): // Wait for the p2p stateprovider to be ready to receive the response.
```

Comment:
```suggestion
			case <-time.After(time.Second):
				// give up if the state provider is not ready for a response
```

Should this have been an error, though?

Reply:
That seems quite reasonable since we effectively _failed_ to send. It seems entirely possible that a caller of this function would be interested in knowing that. Updated.